packages/core-data/src/hooks/test/use-entity-record.js

```diff
@@ -116,22 +116,27 @@ describe( 'useEntityRecord', () => {
 	} );
 
 	it( 'does not resolve entity record when disabled via options', async () => {
-		// Provide response
 		triggerFetch.mockImplementation( () => TEST_RECORD );
 
 		let data;
-		const TestComponent = () => {
-			data = useEntityRecord( 'root', 'widget', 2, {
-				options: { enabled: false },
-			} );
+		const TestComponent = ( { enabled } ) => {
+			data = useEntityRecord( 'root', 'widget', 1, { enabled } );
 			return <div />;
 		};
-		render(
+		const UI = ( { enabled } ) => (
 			<RegistryProvider value={ registry }>
-				<TestComponent />
+				<TestComponent enabled={ enabled } />
 			</RegistryProvider>
 		);
 
+		const { rerender } = render( <UI enabled={ true } /> );
+		await act(
+			() => new Promise( ( resolve ) => setTimeout( resolve, 0 ) )
```

Comment:
It would be nice to add some comments above to explain why we need this and this is the minimal delay we need.

Reply:
What do you think about [9907205](https://github.com/WordPress/gutenberg/pull/56415/commits/9907205ff4258926aafe37b8df80282187cc8f63)?